velox/type/Decimal.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <folly/dynamic.h>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include "velox/common/base/Exceptions.h"
+#include "velox/type/StringView.h"
+
+namespace facebook::velox {
+
+#define PRECISION(typmod) ((uint8_t)(typmod >> 8))
+#define SCALE(typmod) ((uint8_t)(typmod))
+#define TYPMOD(X, Y) ((X << 8) | y)
+#define NUM_DECIMAL_PARAMETERS 2
+
+using int128_t = __int128_t;
+static constexpr uint8_t kMaxPrecisionInt128 = 38;
+static constexpr uint8_t kDefaultScale = 0;
+static constexpr uint8_t kDefaultPrecision = kMaxPrecisionInt128;
+
+/*
+ * This class defines the Velox DECIMAL type support to store
```

Comment:
I believe we decided to use int64_t and int128_t to store the values. Hence, these classes are not needed.

Reply:
I had removed these LongDecimal and ShortDecimal classes. Will add it back.